### PR TITLE
Make FlutterPlatform public so we can start testing/refactoring it

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Makes FlutterPlatform public (though really still an implementation detail) so we can test it. Adds two test cases to verify behavior wrt explicit observatory ports and precompiled entry-points.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
cc @collinjackson

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 1]
```diff
@@ -380,8 +382,9 @@ class _Compiler {
   }
 }
 
-class _FlutterPlatform extends PlatformPlugin {
-  _FlutterPlatform({
+/// The flutter test platform used to integrate with package:test.
+class FlutterPlatform extends PlatformPlugin {
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: Consider adding docs for the public fields/methods.

**Review by zanderso — APPROVED** [review]
lgtm w/ some comments.

File: packages/flutter_tools/lib/src/test/flutter_platform.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/test/flutter_platform.dart]: I'd like to significantly refactor this configuration so these will likely move around a bit over the next few weeks. I will ensure each field is documented once moved

**Pull request merged**